apps/src/templates/teacherDashboard/AssignmentSelector.jsx

```diff
@@ -76,10 +83,17 @@ export default class AssignmentSelector extends Component {
 
   onChangePrimary = (event) => {
     const { assignments, section } = this.props;
-    const currentSecondaryId = assignmentId(null, section.scriptId);
+    let currentSecondaryId;
+    if (!section) {
+      currentSecondaryId = noAssignment;
+    } else {
+      currentSecondaryId = assignmentId(null, section.scriptId);
+    }
```

Comment:
You won't - read a little further on to see how `currentSecondaryId` is used.  It represents the secondary id of the persisted row (not your current edit state) and if it's not found in the current list of selectable secondary ids it gets ignored - so when you change from CSD to CSP you'll have no secondary assignment, because csd1 isn't an option.  However, if you changed back from CSP to CSD we'd reselect csd1 as your default value, since it's what was persisted before you started editing.  You can [read through this test case](https://github.com/code-dot-org/code-dot-org/pull/16401/files/4c499906f59880661780956bea7383997580552f#diff-82c8724e5a2aaa8e2b8be24955179517R98) for an example of this behavior.